app/models/document.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+class Document < ActiveRecord::Base
+
+  validate :validate_event_type, :on => [:create, :update]
+  validate :event_type_already_exist, :on => :create
+
+  EVENTS = [['-- Select --', 0], ['Calendar of events', 'calendar_of_event']].freeze
```

Comment:
Y'know, it would be better if we changed `calendar_of_event` to `calendar_of_events` with an s on the end. Thanks!